src/main/java/com/flippingutilities/utilities/SlotState.java

```diff
@@ -0,0 +1,34 @@
+package com.flippingutilities.utilities;
+
+import com.flippingutilities.model.OfferEvent;
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import net.runelite.api.GrandExchangeOfferState;
+
+import java.time.Instant;
+
+@Data
+@AllArgsConstructor
+public class SlotState {
```

Comment:
Do we want to synchronize this with the API model, or convert it to the DTO before sending it?

Reply:
This was my attempt to make a simple data class that was  equivalent to the DTO. This class is meant to be constructed from the more "raw" internal representation of offer events that the plugin uses. If this doesn't conform to the DTO the api expects, I can change it to do so.

Reply:
Here's what I had planned for the slots.
I think all of the fields are name, but they don't currently share the same variable names just yet.

```typescript
export class CreateGeSlotDto {
  @Min(0)
  @Max(7)
  index: number;

  @IsBoolean()
  isBuyOffer: boolean;

  @IsPositive()
  itemId: number;

  @IsPositive()
  offerQty: number;

  @IsPositive()
  offerPrice: number;

  @IsPositive()
  filledQty: number;

  @IsPositive()
  filledPrice: number;

  @IsOptional()
  @IsDate()
  lastFilledTimestamp: string;

  @IsDate()
  createdAt: string;

  @IsString()
  rsn: string;
}
```